src/search.js

```diff
@@ -0,0 +1,58 @@
+(function() {
+
+  'use strict';
+
+  /* global require */
+
+  var HMS = require('hindley-milner-search');
+
+  //  initSigNodes :: Node -> [Node]
+  function initSigNodes(resultsContainer) {
+    return Array.prototype.reduce.call(
+      document.getElementById('toc').querySelectorAll('li a code'),
+      function(r, _el) {
+        var el = document.createElement('div');
+        el.appendChild(_el.parentNode.cloneNode(true));
+        el.className = 'search-result';
+        var k = el.childNodes[0].getAttribute('href');
+        if (!Object.prototype.hasOwnProperty.call(r.set, k)) {
+          r.set[k] = true;
+          r.uniqs.push(el);
+        }
+        return r;
+      },
+      {set: {}, uniqs: []}
+    ).uniqs;
+  }
+
+  window.addEventListener('DOMContentLoaded', function() {
+    var $container = document.getElementById('yes-matches');
+    var $containerNo = document.getElementById('no-matches');
+    var $els = initSigNodes(document.getElementById('search-results'));
+    var search = HMS.init(Array.prototype.map.call($els, function(el) {
+      return el.innerText.trim().split(/\s+/).join(' ');
+    }), {fuzzy: false}).search;
+    var $prevSearch = '';
```

Comment:
In https://github.com/sanctuary-js/sanctuary-site/commit/6e24452b90e8f4894dbc9e590ed5e51c18795199 I replaced the `state` object with various variables. I assumed each of these variables to be stateful, and used the `$` prefix to remind myself of this.